credsweeper/scanner/scan_type/pem_key_pattern.py

```diff
@@ -41,15 +41,10 @@ def run(cls, config: Config, line: str, line_num: int, file_path: str, rule: Rul
         """
         assert rule.pattern_type == rule.PEM_KEY_PATTERN, \
             "Rules provided to PemKeyPattern.run should have pattern_type equal to PEM_KEY_PATTERN"
-        line_data = cls.get_line_data(config, line, line_num, file_path, rule.patterns[0], rule.filters)
-
-        if line_data is None:
-            return None
 
         if cls.is_pem_key(lines[line_num:]):
-            candidate = Candidate([line_data], rule.patterns, rule.rule_name, rule.severity, config, rule.validations,
-                                  rule.use_ml)
-            return candidate
+            return cls._get_candidate(config, line, line_num, file_path, rule)
```

Comment:
Other option:

```python
            candidate = cls._get_candidate(config, line, line_num, file_path, rule)
            if candidate is not None and cls.is_pem_key(lines[line_num:]):
                return candidate
            return None
```

I'm not sure, but if I remember correctly `get_line_data` would perform much faster than `is_pem_key`
So this order might be better

I want to test speed in both cases and will post results in an hour

Reply:
It seems that difference between implementations is insignificant and less than 1%
Tested on CredData and few other repos